tests/consensus_tests/test_jwt_validation.py

```diff
@@ -92,66 +570,1124 @@ def scroll_with_token(uri: str, collection: str, token: str) -> requests.Respons
     with pytest.raises(requests.HTTPError):
-        scroll_with_token(uri, COLL_NAME, token)
+        scroll_with_token(COLL_NAME, token)
 
     # Create collection
-    create_validation_collection(
-        uri, secondary_collection, shard_number=N_SHARDS, replication_factor=N_REPLICA
-    )
+    create_validation_collection(validation_collection)
 
     # Check it does not work now
     with pytest.raises(requests.HTTPError):
-        res = scroll_with_token(uri, COLL_NAME, token)
+        res = scroll_with_token(COLL_NAME, token)
 
     # Upload validation point
     res = requests.put(
-        f"{uri}/collections/{secondary_collection}/points?wait=true",
+        f"{REST_URI}/collections/{validation_collection}/points?wait=true",
         json={
             "points": [
                 {
                     "id": 42,
-                    "vector": [0],
+                    "vectors": {},
                     "payload": {key: value},
                 }
             ]
         },
         headers=API_KEY_HEADERS,
     )
-    assert_http_ok(res)
+    res.raise_for_status()
 
     # Check that token works now
-    res = scroll_with_token(uri, COLL_NAME, token)
+    res = scroll_with_token(COLL_NAME, token)
     assert len(res.json()["result"]["points"]) == 10
 
     # Delete validation point
     res = requests.post(
-        f"{uri}/collections/{secondary_collection}/points/delete?wait=true",
+        f"{REST_URI}/collections/{validation_collection}/points/delete?wait=true",
         json={"points": [42]},
         headers=API_KEY_HEADERS,
     )
     res.raise_for_status()
 
     # Check it does not work now
     with pytest.raises(requests.HTTPError):
-        scroll_with_token(uri, COLL_NAME, token)
+        scroll_with_token(COLL_NAME, token)
+
+    fixtures.drop_collection(REST_URI, validation_collection, headers=API_KEY_HEADERS)
+    fixtures.drop_collection(REST_URI, validation_collection, headers=API_KEY_HEADERS)
+
+
+def check_rest_access(
+    method: str,
+    path: str,
+    body: Optional[Union[dict, Callable[[], dict]]],
+    should_succeed: bool,
+    token: str,
+    path_params: dict = {},
+    request_kwargs: dict = {},
+):
+    if isfunction(body):
+        body = body()
+
+    concrete_path_params = {}
+    for key, value in path_params.items():
+        concrete_path_params[key] = value() if isfunction(value) else value
+
+    path = path.format(**concrete_path_params)
+
+    res = requests.request(
+        method,
+        f"{REST_URI}{path}",
+        headers={"authorization": f"Bearer {token}"},
+        json=body,
+        **request_kwargs,
+    )
+
+    if should_succeed:
+        assert res.status_code < 500 and res.status_code not in [
+            401,
+            403,
+        ], f"{method} {path} failed with {res.status_code}: {res.text}"
+    else:
+        assert res.status_code in [
+            401,
+            403,
+        ], f"{method} {path} failed with {res.status_code}: {res.text}"
```

Comment:
No need to check for 401 Unauthorized, this is for wrong/missing creds.
Check only for 403 Forbidden.
You are the one who suggested to distinguish them in the first place. 😅

Also, I suggest to reword the latter assertion message to indicate that it is *not* failed (but it should).

```suggestion
    if should_succeed:
        assert (
            res.status_code < 500 and res.status_code != 403
        ), f"{method} {path} failed with {res.status_code}: {res.text}"
    else:
        assert (
            res.status_code != 403
        ), f"{method} {path} was not forbidden with {res.status_code}: {res.text}"
```

Reply:
I already removed the 401 check in #4008, I think it is fine to merge this as is to save some conflict resolution afterwards :)

I agree with the rewording, though. I'll edit the message in #4008 too.